File: Week 2/Problem Set 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float L = (count_letters(text) / (float) count_words(text)) * 100;
    float S = (count_sentences(text) / (float) count_words(text)) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        if (index > 16)
        {
            printf("Grade 16+\n");
        }
        else
        {
            printf("Grade %.0f\n", round(index));
        }
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 32)
        {
            words++;
        }
    }
    //+1 for the last word in the text
    return words + 1;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Week 6/Problem Set 6/sentimental-readability/readability.py
def main():
    str = input("Text: ")
    L = (count_letters(str) / count_words(str)) * 100
    S = (count_sentences(str) / count_words(str)) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


def count_letters(str):
    num = 0
    for letter in str:
        if letter.isalpha():
            num += 1
    return num


def count_words(str):
    return str.count(" ") + 1


def count_sentences(str):
    num = 0
    for letter in str:
        if letter in [".", "!", "?"]:
            num += 1
    return num


if __name__ == "__main__":
    main()
